Tools/LyTestTools/ly_test_tools/o3de/editor_test.py

```diff
@@ -921,33 +253,34 @@ def _exec_editor_test(self, request: _pytest.fixtures.FixtureRequest,
                     else:
                         logger.warning(f"Crash occurred, but could not find log {crash_file_name}")
                 else:
-                    test_result = Result.Fail(test_spec, output, editor_log_content)
+                    test_result = Result.Fail(EditorTestSuite._log_attribute, test_spec, output, editor_log_content)
         except WaitTimeoutError:
             output = editor.get_output()
             editor.stop()
             editor_log_content = editor_utils.retrieve_editor_log_content(run_id, log_name, workspace)
-            test_result = Result.Timeout(test_spec, output, test_spec.timeout, editor_log_content)
+            test_result = Result.Timeout(EditorTestSuite._log_attribute, test_spec, output, test_spec.timeout, editor_log_content)
     
         editor_log_content = editor_utils.retrieve_editor_log_content(run_id, log_name, workspace)
-        results = self._get_results_using_output([test_spec], output, editor_log_content)
+        results = self._get_results_using_output(EditorTestSuite._log_attribute, [test_spec], output, editor_log_content)
         results[test_spec.__name__] = test_result
         return results
 
-    def _exec_editor_multitest(self, request: _pytest.fixtures.FixtureRequest,
-                               workspace: ly_test_tools._internal.managers.workspace.AbstractWorkspaceManager,
-                               editor: ly_test_tools.launchers.platforms.base.Launcher, run_id: int, log_name: str,
-                               test_spec_list: list[EditorTest],
+    def _exec_editor_multitest(self,
```

Comment:
Most of the remaining 400 lines of this file appears unrelated to running in specifically the editor, and more like framework-level orchestration of running an application. I would expect to see it in multi_test_framework.py and not duplicated into both editor_test and material_editor_test.

Perhaps there can be a parent function which each of these call with a few parameters? Or a parent function which expects to call a function implented in each child that which performs the few unique steps?

Reply:
I have successfully moved the executor code over to `multi_test_framework.py` and it works for Editor tests. However, there are still a few bugs i need to work out with the MaterialEditor tests before it can also utilize `multi_test_framework.py`.

I will push the changes I have so far, but ignore `MaterialEditor` until I push again as it has commented out executor code that needs to work between both Editor and MaterialEditor.